apps/extension/src/provider/InjectedAnoma.ts

```diff
@@ -1,176 +1,83 @@
-      }
-    });
+  public async connect(chainId: string): Promise<void> {
+    return await InjectedProxy.requestMethod<string, void>("connect", chainId);
   }
 
-  protected requestMethod(method: ProxyMethod, args: any): Promise<any> {
-    const bytes = new Uint8Array(8);
-    const id: string = Array.from(crypto.getRandomValues(bytes))
-      .map((value) => {
-        return value.toString(16);
-      })
-      .join("");
-
-    const proxyMessage: ProxyRequest = {
-      type: ProxyRequestTypes.Request,
-      id,
-      method,
-      args,
-    };
-
-    return new Promise((resolve, reject) => {
-      const receiveResponse = (e: MessageEvent) => {
-        const proxyResponse: ProxyRequestResponse = this.parseMessage
-          ? this.parseMessage(e.data)
-          : e.data;
-
-        if (
-          !proxyResponse ||
-          proxyResponse.type !== ProxyRequestTypes.Response
-        ) {
-          return;
-        }
-
-        if (proxyResponse.id !== id) {
-          return;
-        }
-
-        this.eventListener.removeMessageListener(receiveResponse);
-
-        const { result } = proxyResponse;
-
-        if (!result) {
-          reject(new Error("Result is null"));
-          return;
-        }
-
-        if (result.error) {
-          reject(new Error(result.error as any));
-          return;
-        }
+  public async chain(chainId: string): Promise<Chain | undefined> {
+    return InjectedProxy.requestMethod<string, Chain>("chain", chainId);
+  }
 
-        resolve(result.return);
-      };
+  public async chains(): Promise<Chain[]> {
+    return await InjectedProxy.requestMethod<void, Chain[]>(
+      "chains",
+      undefined
+    );
+  }
 
-      this.eventListener.addMessageListener(receiveResponse);
-      this.eventListener.postMessage(proxyMessage);
-    });
+  public async suggestChain(chain: Chain): Promise<void> {
+    await InjectedProxy.requestMethod<Chain, void>("suggestChain", chain);
   }
 
-  constructor(
-    private readonly _version: string,
-    protected readonly eventListener: {
-      addMessageListener: (fn: (e: any) => void) => void;
-      removeMessageListener: (fn: (e: any) => void) => void;
-      postMessage: (message: any) => void;
-    } = {
-      addMessageListener: (fn: (e: any) => void) =>
-        window.addEventListener("message", fn),
-      removeMessageListener: (fn: (e: any) => void) =>
-        window.removeEventListener("message", fn),
-      postMessage: (message) =>
-        window.postMessage(message, window.location.origin),
-    },
-    protected readonly parseMessage?: (message: any) => any
-  ) {}
+  public async accounts(chainId: string): Promise<DerivedAccount[]> {
+    return await InjectedProxy.requestMethod<string, DerivedAccount[]>(
+      "accounts",
+      chainId
+    );
+  }
 
-  public async connect(chainId: string): Promise<void> {
-    return await this.requestMethod("connect", chainId);
+  public getSigner(chainId: string): ISigner<Account> | undefined {
```

Comment:
The `Account` type should only be a subset of `DerivedAccount`. This is because an account returned to the interface should only have _some_ of the fields, such as alias, address, and likely public-key at some point, but shouldn't return anything else. The signer will only return those few fields to the interface, not the full stored account data.

Reply:
@memasdeligeorgakis Also, that is correct that `namada-interface/packages/tx/src/Account.ts` is unrelated to the extension. That package is used by `namada-interface`, and until the extension is integrated, that will need to remain.